utils/generate-api.py

```diff
@@ -130,11 +137,136 @@ def sort(self):
+
+        header_separator = "# -----------------------------------------------------"
+        License_header_end_1 = "# GitHub history for details."
+        License_header_end_2 = "#  under the License."
+
+        update_header = True
+        License_position = 0
+        self.header = "\n".join(
+            line for line in self.header.split("\n") if "from .utils import" not in line
+        )
+        if os.path.exists(self.filepath):
+            with open(self.filepath, "r") as f:
+                content = f.read()
+            if header_separator in content:
+                update_header = False
+                header_end_position = (
+                    content.find(header_separator) + len(header_separator) + 2
+                )
+                header_position = content.rfind("\n", 0, header_end_position) + 1
+            if License_header_end_1 in content:
+                if License_header_end_2 in content:
+                    position = (
+                        content.find(License_header_end_2)
+                        + len(License_header_end_2)
+                        + 2
+                    )
+                else:
+                    position = (
+                        content.find(License_header_end_1)
+                        + len(License_header_end_1)
+                        + 2
+                    )
+                License_position = content.rfind("\n", 0, position) + 1
+
         with open(self.filepath, "w") as f:
-            f.write(self.header)
+            if update_header is True:
+                f.write(self.header[:License_position])
+                f.write("\n")
+                f.write("# ----------------------------------------------------\n")
+                f.write("# THIS CODE IS GENERATED. MANUAL EDITS WILL BE LOST. \n")
+                f.write("\n")
+                f.write(
+                    '# To contribute, please make necessary modifications to either "Python generator":\n'
+                )
+                f.write(
+                    "# https://github.com/opensearch-project/opensearch-py/blob/main/utils/generate-api.py\n"
+                )
+                f.write('# or "OpenAPI specs":\n')
+                f.write(
+                    "# https://github.com/opensearch-project/opensearch-api-specification/blob/main/OpenSearch.openapi.json\n"
+                )
+                f.write("# -----------------------------------------------------\n")
+                f.write("\n")
+                f.write("#replace_token#\n")
+                f.write(self.header[License_position:])
+            else:
+                f.write(self.header[:header_position])
+                f.write("\n")
+                f.write("#replace_token#\n")
+                f.write(self.header[header_position:])
             for api in self._apis:
                 f.write(api.to_python())
 
+        # Added "delete point in time" API manually to prevent breaking changes
```

Comment:
Yikes, this seems like a huge hack. We need to find a better way, for example by putting generated APIs into `generated/` and adding `patches/` folder that includes those APIs that should not be generated. You can use file names to decide which one needs to be which.